### PR TITLE
fix(theme): applied tw nested group 

### LINKED ISSUE #2324: [BUG] - Multiple Selection of the table
### NextUI Version

2.2.9

### Describe the bug

Hi, I have a problem related to the selectionMode = "multiple" option of the table, what this does is to generate a checkbox to select multiple rows, but I have in my custom cells a radio button and a checkbox, of which if I press any checkbox of those that are generated by the option "multiple selection" these visually activate the checkbox or radio button in the table row, I checked that it was only visual and not that it was really making them check, but I do not see or I did not find way to make that event of click in the checkboxes not to affect to what I had on my customized cells of my table and prevent that situation. 

- **I think it would be good to have a new section in the table documentation, where you can add some input or checkbox as a custom cell and use at the same time the "selectionMode ="multiple"" checkbox.**

### Your Example Website or App

_No response_

### Steps to Reproduce the Bug or Issue

1. Create table. 
2. Add the property "selectionMode ="multiple", to add the checkboxes. 
3. Add custom cells on the table and use checkbox or radio buttons from nextUI. 
4. if you click on the checkbox of the selectionMode you will see the other checkbox or radio buttons being selected.

### Expected behavior

That the checkbox of the table can be controlled more thoroughly, and that in the components if I put custom cells a checkbox does not affect another one or a radius does not affect another one.

### Screenshots or Videos

Screenshot 2024-02-07 144136
Screenshot 2024-02-07 144210

### Operating System Version

Windows

### Browser

Chrome

### LINKED ISSUE #2959: [BUG] - Switch component inside of table is trigger when a row is selected
### NextUI Version

2.3.6

### Describe the bug

I have a products table:
image

When the switch is set active, there's no problem when selecting the row, however when the switch is inactive whenever I select the row it toggles the switch.
[scrnli_5_7_2024_10-32-43 AM.webm](https://github.com/nextui-org/nextui/assets/43927089/6ac9f77c-0cf0-4805-aae7-ff94bcde8c44)

### Your Example Website or App

_No response_

### Steps to Reproduce the Bug or Issue

1. Create a table component
2. Add a Switch component to the row
3. Set it active on page load using useState
4. Click on the row - nothing will happen.
5. Toggle it to inactive
6. Click on the row and it will toggle with the row selection

### Expected behavior

I expected it to not change on row selection no matter what the state of the switch is but what I'm seeing is a change.

### Screenshots or Videos

_No response_

### Operating System Version

Windows

### Browser

Chrome

### LINKED ISSUE #3544: [BUG] - Disabled checkboxes in selectable table do not behaves as expected
### NextUI Version

2.4.6

### Describe the bug

When I use a `<Table>` with a non-undefined `selectionMode`, all the checkboxed with `isDisabled` or `isReadonly` switch state when clicking on a table row.

Here is athe code of my simple component:

```javascript
"use client";

import {
  Table,
  TableBody,
  TableCell,
  TableColumn,
  TableHeader,
  TableRow,
} from "@nextui-org/table";
import { Checkbox } from "@nextui-org/checkbox";

export default function Home() {
  return (
    <section>
      <h1>Selectable table: KO</h1>
      <Table selectionMode="single">
        <TableHeader>
          <TableColumn>Disabled</TableColumn>
          <TableColumn>Readonly</TableColumn>
        </TableHeader>
        <TableBody>
          <TableRow>
            <TableCell>
              <Checkbox isDisabled>Disabled</Checkbox>
            </TableCell>
            <TableCell>
              <Checkbox isReadOnly>Readonly</Checkbox>
            </TableCell>
          </TableRow>
        </TableBody>
      </Table>

      <h1>Unselectable table: OK</h1>
      <Table>
        <TableHeader>
          <TableColumn>Disabled</TableColumn>
          <TableColumn>Readonly</TableColumn>
        </TableHeader>
        <TableBody>
          <TableRow>
            <TableCell>
              <Checkbox isDisabled>Disabled</Checkbox>
            </TableCell>
            <TableCell>
              <Checkbox isReadOnly>Readonly</Checkbox>
            </TableCell>
          </TableRow>
        </TableBody>
      </Table>
    </section>
  );
}
```

### Your Example Website or App

https://github.com/nbusser/nextui-issue-checkbox-tables

### Steps to Reproduce the Bug or Issue

1. `pnpm i`
2. `npm run dev`
3. Go on `localhost:3000`

### Expected behavior

As a user, I expect being NOT able to change the state of a disabled/readonly checkbox within a table

### Screenshots or Videos

https://github.com/user-attachments/assets/334dde76-96d3-44f2-8208-d9a88edd4336

### Operating System Version

macOs

### Browser

Firefox

### PR DESCRIPTION
**Pull request by wingkwong**
- closes #3544
- closes #2324
- closes #2959

## 📝 Description

Applied nested group to components so that styles from other components won't mess up with others. 

## ⛳️ Current behavior (updates)

let's say there is a switch component inside a table row with select button. clicking select button would also toggle the switch. The reason is that when we click select, `data-selected` for select will be set to `true`. However, all `group-data-[selected=true]:*` in switch component will be applied as well since all components share the same parent `group`.

[pr3658-before-demo.webm](https://github.com/user-attachments/assets/6d3af46d-f287-4c4a-9881-f322e49c1d15)

## 🚀 New behavior

[pr3658-after-demo.webm](https://github.com/user-attachments/assets/083b872e-8d65-4f80-97f4-d8257daa926f)

## Summary by CodeRabbit

- **New Features**
	- Introduced a new table template with toggle functionality for active status, enhancing interactivity.
	- Enhanced styling for various components including autocomplete, checkbox, date input, date picker, input, radio, select, and toggle to improve clarity and maintainability.

- **Bug Fixes**
	- Addressed styling issues related to nested groups in the theme framework, ensuring better visual consistency and functionality.

- **Chores**
	- Updated class naming conventions across multiple components for improved specificity in styling.

### CODE REVIEW

**Comment by ryo-manba** [comment]
Hey @wingkwong, could you please fix the conflicts?

**Comment by wingkwong** [comment]
@ryo-manba updated. thanks.

File: .changeset/clean-teachers-heal.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+"@nextui-org/theme": patch
+---
+
+applied tw nested group (#3544, #2324)
```
Comment (ryo-manba) [inline .changeset/clean-teachers-heal.md:5]: Please add #2959.

**Review by ryo-manba — COMMENTED** [review]
Could you add a test or Storybook for me since the changes are extensive?

**Comment by ryo-manba** [comment]
@wingkwong 
Will this issue also be fixed?
- https://github.com/nextui-org/nextui/issues/3725

**Comment by wingkwong** [comment]
> @wingkwong Will this issue also be fixed?
> 
> * [[BUG] - Table selection - Unintuitive behaviour #3725](https://github.com/nextui-org/nextui/issues/3725)

Just tested with PR docs and still got same result. I don't think it's not related to this PR.

**Review by ryo-manba — APPROVED** [review]
Thanks!

**Pull request merged**